nx/lib/nx/serving.ex

```diff
@@ -516,44 +516,48 @@ defmodule Nx.Serving do
   def handle_info({__MODULE__, :batched_run, ref, %Nx.Batch{} = batch}, state) do
     %{limit: limit, stack: {_, count}} = state
 
-    state =
-      cond do
-        # Single entry takes the whole batch.
-        # Execute what we have (if any) and execute a new one.
-        batch.size == limit ->
-          state
-          |> server_execute()
-          |> server_stack(ref, batch)
-          |> server_execute()
-
-        # First entry in batch.
-        count == 0 ->
-          state
-          |> server_timer()
-          |> server_stack(ref, batch)
-
-        # We don't exceed the limit.
-        batch.size + count < limit ->
-          server_stack(state, ref, batch)
-
-        # We go over the limit.
-        batch.size + count > limit ->
-          {current, next} = Nx.Batch.split(batch, limit - count)
-
-          state
-          |> server_stack(ref, current)
-          |> server_execute()
-          |> server_timer()
-          |> server_stack(ref, next)
-
-        # Exact match.
-        true ->
-          state
-          |> server_stack(ref, batch)
-          |> server_execute()
-      end
+    meta = %{count: count, limit: limit, module: state.module, name: state.name}
+
+    :telemetry.span([:nx, :serving, :batched_run], meta, fn ->
```

Comment:
Unfortunately this event is not very helpful because `server_execute` only starts a task. :( So if we want to measure it from the server, we cannot use span but emulate span, which is somewhat annoying. The other thing is that we would want to include the queue size as a measurement, not as metadata, and we cannot add new measurements to telemetry spans anyway.

So my suggestion, for now, is to start with the client events? And then later we figure out the queue size. Because for queue size we will need to either send it to the client (and use custom spans) or emit additional events on the server (spans or not).

Reply:
Ok, thanks for the context. I'll roll this back and switch to client events initially.